src/test/groovy/ee/tuleva/onboarding/epis/EpisServiceIntSpec.groovy

```diff
@@ -92,6 +106,75 @@ class EpisServiceIntSpec extends Specification {
     applications.first().documentNumber == "123456"
   }
 
+  def "getApplications - only one request per person is allowed to run at any time"() {
+    given:
+    mockServerClient
+        .when(request("/applications")
+            .withHeader("Authorization", "Bearer dummy"))
+        .respond(response()
+            .withDelay(TimeUnit.MILLISECONDS, 500)
+            .withContentType(MediaType.APPLICATION_JSON)
+            .withBody("""
+                        [{
+                         "currency": "EUR",
+                         "amount": 100.0,
+                         "status": "PENDING",
+                         "type": "TRANSFER",
+                         "id": 123,
+                         "documentNumber": "123456"
+                        }]
+                        """))
+    when:
+    ExecutorService executor = Executors.newFixedThreadPool(2)
+    with(executor) {
+      submit({
+        setUpSecurityContext()
+        episService.getApplications(PersonFixture.samplePerson())
+        SecurityContextHolder.clearContext()
+      })
+      submit({
+        setUpSecurityContext()
+        episService.getApplications(PersonFixture.samplePerson())
+        SecurityContextHolder.clearContext()
+      })
+      shutdown()
+      awaitTermination(5, TimeUnit.SECONDS)
```

Comment:
Does this make the test slow?

Reply:
No it doesn't wait for 5 seconds, it just times out if for some unknown reason it would take 5s.